Cubical/Data/Equality.agda

```diff
@@ -1,50 +1,331 @@
+equivToEquivPath (f , p) =
+  (f , λ { .equiv-proof y → helper2 fiberPathToFiber fiberToFiberPath fiberPathToFiberPath (p .equiv-proof y) })
+
+-- Go from a Path equivalence to an ≡ equivalence
+equivPathToEquiv : EquivPath A B → A ≃ B
+equivPathToEquiv (f , p) =
+  (f , λ { .equiv-proof y → helper1 fiberPathToFiber fiberToFiberPath fiberToFiber (p .equiv-proof y) })
+
+equivToEquiv : {A B : Type ℓ} (p : A ≃ B) → Path _ (equivPathToEquiv (equivToEquivPath p)) p
+equivToEquiv (f , p) i =
+  (f , isPropIsEquiv (λ { .equiv-proof y → helper1 fiberPathToFiber fiberToFiberPath fiberToFiber
+                                             (helper2 fiberPathToFiber fiberToFiberPath fiberPathToFiberPath (p .equiv-proof y)) }) p i)
+
+equivPathToEquivPath : {A B : Type ℓ} (p : EquivPath A B) → Path _ (equivToEquivPath (equivPathToEquiv p)) p
+equivPathToEquivPath (f , p) i =
+  ( f , isPropIsEquivPath f (equivToEquivPath (equivPathToEquiv (f , p)) .snd) p i )
+
+equivPath≡Equiv : {A B : Type ℓ} → Path _ (EquivPath A B) (A ≃ B)
+equivPath≡Equiv {ℓ} = isoToPath (iso (equivPathToEquiv {ℓ}) equivToEquivPath equivToEquiv equivPathToEquivPath)
+
+path≡Eq : {A B : Type ℓ} → Path _ (Path _ A B) (A ≡ B)
+path≡Eq = isoToPath (iso pathToEq eqToPath pathToEq-eqToPath eqToPath-pathToEq)
+
+-- Univalence formulated using ≡ everywhere
+univalenceEq : (A ≡ B) ≃ (A ≃ B)
+univalenceEq {A = A} {B = B} = equivPathToEquiv rem
+  where
+  rem0 : Path _ (Lift (EquivPath A B)) (Lift (A ≃ B))
+  rem0 = congPath Lift equivPath≡Equiv
+
+  rem1 : Path _ (A ≡ B) (Lift (A ≃ B))
+  rem1 i = hcomp (λ j → λ { (i = i0) → path≡Eq {A = A} {B = B} j
+                          ; (i = i1) → rem0 j })
+                 (univalencePath {A = A} {B = B} i)
+
+  rem : EquivPath (A ≡ B) (A ≃ B)
+  rem = compEquiv (eqweqmap rem1) (invEquiv LiftEquiv)
+
+
+-- Propositional truncation using ≡ with paths under the hood
+
+∥∥-isProp : ∀ (x y : ∥ A ∥) → x ≡ y
+∥∥-isProp x y = pathToEq (squashPath x y)
+
+∥∥-recursion : ∀ {A : Type ℓ} {P : Type ℓ} → isProp P → (A → P) → ∥ A ∥ → P
+∥∥-recursion Pprop = recPropTruncPath (isPropToIsPropPath Pprop)
+
+∥∥-induction : ∀ {A : Type ℓ} {P : ∥ A ∥ → Type ℓ} → ((a : ∥ A ∥) → isProp (P a)) →
+                ((x : A) → P ∣ x ∣) → (a : ∥ A ∥) → P a
+∥∥-induction Pprop = elimPropTruncPath (λ a → isPropToIsPropPath (Pprop a))
+
+
+-- The circle using ≡
+
+loop : base ≡ base
+loop = pathToEq loopPath
+
+S¹-rec : {A : Type ℓ} (b : A) (l : b ≡ b) → S¹ → A
+S¹-rec b l = S1.rec b (eqToPath l)
+
+S¹-elim : (C : S¹ → Type ℓ) (b : C base) (l : transport C loop b ≡ b) → (x : S¹) → C x
+S¹-elim C b l x = S1.ind C b (eqToPath (substPath≡transport C b loopPath ∙ l)) x
+
+S¹-recβ : (b : A) (l : b ≡ b) → ap (S¹-rec b l) loop ≡ l
+S¹-recβ b l =
+  ap (S¹-rec b l) loop ≡⟨ ap≡congPath (S¹-rec b l) loopPath ⟩
+  pathToEq (congPath (S¹-rec b l) loopPath) ≡⟨ refl ⟩
+  pathToEq (eqToPath l) ≡⟨ pathToEq (pathToEq-eqToPath l) ⟩
+  l ∎
+
+
+-- This seems hard to prove?
```

Comment:
@Saizan : any ideas for how to prove this one?

Reply:
Seems like you need an heterogeneous version of `ap≡congPath`, with a corresponding `pathToEqOver`.

Reply:
This looks like the kind of thing I cover in my paper [Higher Inductive Type Eliminators Without Paths](http://www.cse.chalmers.se/~nad/publications/danielsson-hits-without-paths.html), which supports "arbitrary" notions of equality with J. I would guess that you could finish the proof easily if you import my library and change the definition of `S¹-elim`.

Reply:
My [proof](http://www.cse.chalmers.se/~nad/listings/equality/Circle.html#elim-loop) of the corresponding property.